crates/anvil/core/src/eth/block.rs

```diff
@@ -36,8 +34,8 @@ impl Block {
         T: Into<Transaction>,
     {
         let transactions: Vec<_> = transactions.into_iter().map(Into::into).collect();
-        let transactions_root =
-            trie::ordered_trie_root(transactions.iter().map(|r| r.encoded_2718()));
+        let encoded_transactions: Vec<_> = transactions.iter().map(|r| r.encoded_2718()).collect();
+        let transactions_root = root::ordered_trie_root(&encoded_transactions);
```

Comment:
here we can actually use:

https://github.com/alloy-rs/alloy/blob/833f9a22ca7c30079d64554d435561877c15267a/crates/consensus/src/proofs.rs#L19-L19